<a href="https://colab.research.google.com/github/salacamifda/Altura-Promedio-Olimpico/blob/main/Altura_Promedio_Olimpico.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Tarea

1. Carguen este conjunto de datos y explórenlo. Respondan las siguientes preguntas:

     a. ¿Cuántas filas hay?

     b. ¿Cuántas columnas hay?

     c. ¿A qué columnas le faltan datos? (¿Por qué creen que hay muchos valores nulos en la columna Medals (medallas)?

     d. ¿Cuántas entradas corresponden a la ciudad de Londres?

     e. ¿Cuántos años tiene el atleta más joven en nuestros datos de muestra?  Pista: usen min()

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('/content/athleteEventsNoPersonal.csv')

df.head()

ID   Age  Height         Team  NOC        Games  Year  Season         City  \
0   5  21.0   185.0  Netherlands  NED  1988 Winter  1988  Winter      Calgary   
1   5  21.0   185.0  Netherlands  NED  1988 Winter  1988  Winter      Calgary   
2   5  25.0   185.0  Netherlands  NED  1992 Winter  1992  Winter  Albertville   
3   5  25.0   185.0  Netherlands  NED  1992 Winter  1992  Winter  Albertville   
4   5  27.0   185.0  Netherlands  NED  1994 Winter  1994  Winter  Lillehammer   

           Sport                               Event Medal  
0  Speed Skating    Speed Skating Women's 500 metres   NaN  
1  Speed Skating  Speed Skating Women's 1,000 metres   NaN  
2  Speed Skating    Speed Skating Women's 500 metres   NaN  
3  Speed Skating  Speed Skating Women's 1,000 metres   NaN  
4  Speed Skating    Speed Skating Women's 500 metres   NaN

In [3]:
df

ID   Age  Height         Team  NOC        Games  Year  Season  \
0           5  21.0   185.0  Netherlands  NED  1988 Winter  1988  Winter   
1           5  21.0   185.0  Netherlands  NED  1988 Winter  1988  Winter   
2           5  25.0   185.0  Netherlands  NED  1992 Winter  1992  Winter   
3           5  25.0   185.0  Netherlands  NED  1992 Winter  1992  Winter   
4           5  27.0   185.0  Netherlands  NED  1994 Winter  1994  Winter   
...       ...   ...     ...          ...  ...          ...   ...     ...   
40611  135567  28.0   183.0       Russia  RUS  2004 Summer  2004  Summer   
40612  135570  27.0   176.0       Poland  POL  2014 Winter  2014  Winter   
40613  135570  27.0   176.0       Poland  POL  2014 Winter  2014  Winter   
40614  135571  30.0   185.0       Poland  POL  1998 Winter  1998  Winter   
40615  135571  34.0   185.0       Poland  POL  2002 Winter  2002  Winter   

                 City          Sport  \
0             Calgary  Speed Skating   
1             Calgary  Speed Skating   
2         Albertville  Speed Skating   
3         Albertville  Speed Skating   
4         Lillehammer  Speed Skating   
...               ...            ...   
40611          Athina         Rowing   
40612           Sochi    Ski Jumping   
40613           Sochi    Ski Jumping   
40614          Nagano      Bobsleigh   
40615  Salt Lake City      Bobsleigh   

                                          Event Medal  
0              Speed Skating Women's 500 metres   NaN  
1            Speed Skating Women's 1,000 metres   NaN  
2              Speed Skating Women's 500 metres   NaN  
3            Speed Skating Women's 1,000 metres   NaN  
4              Speed Skating Women's 500 metres   NaN  
...                                         ...   ...  
40611    Rowing Men's Lightweight Coxless Fours   NaN  
40612  Ski Jumping Men's Large Hill, Individual   NaN  
40613        Ski Jumping Men's Large Hill, Team   NaN  
40614                      Bobsleigh Men's Four   NaN  
40615                      Bobsleigh Men's Four   NaN  

[40616 rows x 12 columns]

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40616 entries, 0 to 40615
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   ID      40616 non-null  int64  
 1   Age     39143 non-null  float64
 2   Height  31615 non-null  float64
 3   Team    40616 non-null  object 
 4   NOC     40616 non-null  object 
 5   Games   40616 non-null  object 
 6   Year    40616 non-null  int64  
 7   Season  40616 non-null  object 
 8   City    40616 non-null  object 
 9   Sport   40616 non-null  object 
 10  Event   40616 non-null  object 
 11  Medal   5917 non-null   object 
dtypes: float64(2), int64(2), object(8)
memory usage: 3.7+ MB


In [5]:
df.isna()


ID    Age  Height   Team    NOC  Games   Year  Season   City  Sport  \
0      False  False   False  False  False  False  False   False  False  False   
1      False  False   False  False  False  False  False   False  False  False   
2      False  False   False  False  False  False  False   False  False  False   
3      False  False   False  False  False  False  False   False  False  False   
4      False  False   False  False  False  False  False   False  False  False   
...      ...    ...     ...    ...    ...    ...    ...     ...    ...    ...   
40611  False  False   False  False  False  False  False   False  False  False   
40612  False  False   False  False  False  False  False   False  False  False   
40613  False  False   False  False  False  False  False   False  False  False   
40614  False  False   False  False  False  False  False   False  False  False   
40615  False  False   False  False  False  False  False   False  False  False   

       Event  Medal  
0      False   True  
1      False   True  
2      False   True  
3      False   True  
4      False   True  
...      ...    ...  
40611  False   True  
40612  False   True  
40613  False   True  
40614  False   True  
40615  False   True  

[40616 rows x 12 columns]

In [6]:
df.isna().sum()


ID            0
Age        1473
Height     9001
Team          0
NOC           0
Games         0
Year          0
Season        0
City          0
Sport         0
Event         0
Medal     34699
dtype: int64



*   A). son 40616 FIlas
*   B). Son 12 columnas
*   c). Hay muchos valores nulos en la columna medals puesto que no todos los deportistas ganaron medallas
*   D).entradas corresponden a la ciudad de Londres:




In [8]:
city_filter = df['City']. value_counts()
 
city_filter

London                    3370
Athina                    2325
Beijing                   2056
Sydney                    2049
Atlanta                   2037
Rio de Janeiro            2037
Barcelona                 1973
Los Angeles               1864
Seoul                     1852
Munich                    1525
Helsinki                  1305
Roma                      1259
Montreal                  1253
Mexico City               1146
Moskva                    1116
Tokyo                     1087
Paris                     1064
Berlin                     907
Sochi                      787
Amsterdam                  757
Torino                     687
Vancouver                  673
Melbourne                  670
Antwerpen                  645
Salt Lake City             639
Stockholm                  600
Nagano                     586
Albertville                545
Lillehammer                541
Innsbruck                  504
Calgary                    393
Sarajevo                   331
Lake Pla

In [10]:
london_filter = df['City']== 'London'

london_filter. value_counts()

False    37246
True      3370
Name: City, dtype: int64



*   D). Entradas para Londres son 3370.
*   e. ¿Cuántos años tiene el atleta más joven en nuestros datos de muestra? Pista: usen min()



In [11]:
age_filter = df['Age']. value_counts()
 
age_filter

23.0    3286
24.0    3173
22.0    3112
25.0    2920
21.0    2846
        ... 
67.0       3
69.0       3
11.0       3
73.0       2
68.0       1
Name: Age, Length: 64, dtype: int64

In [16]:
minvalue = df['Age'].min()

minvalue

11.0

El atleta más joven de las olimpiadas tiene 11 años.

2. Creen filtros para encontrar:

     a. ¿Cuántos atletas que participaron en el deporte "Tug-Of-War" (tira y afloja) pertenecían en el equipo “Sweden” (sueco) en el conjunto de datos de muestra?

In [19]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40616 entries, 0 to 40615
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   ID      40616 non-null  int64  
 1   Age     39143 non-null  float64
 2   Height  31615 non-null  float64
 3   Team    40616 non-null  object 
 4   NOC     40616 non-null  object 
 5   Games   40616 non-null  object 
 6   Year    40616 non-null  int64  
 7   Season  40616 non-null  object 
 8   City    40616 non-null  object 
 9   Sport   40616 non-null  object 
 10  Event   40616 non-null  object 
 11  Medal   5917 non-null   object 
dtypes: float64(2), int64(2), object(8)
memory usage: 3.7+ MB


In [22]:
tug_of_war_filter = df['Sport']== 'Tug-Of-War'

tug_of_war_filter. value_counts()

False    40598
True        18
Name: Sport, dtype: int64

In [24]:
sweden_filter = df['Team']== 'Sweden'
sweden_filter.value_counts()

False    39484
True      1132
Name: Team, dtype: int64

In [26]:
t_s_df = tug_of_war_filter & sweden_filter

t_s_df.value_counts()

# 6 equipos del deporte tug_of_war partipicaron en el equipo Sweden

False    40610
True         6
dtype: int64

b. ¿Cuántas medallas de oro se le entregaron a los atletas que tenían más de 40 años?

In [33]:
over_40_filter = df['Age'] > 40
over_40_filter.value_counts()

False    39413
True      1203
Name: Age, dtype: int64

In [34]:
goldenMedals_filter = df['Medal']=='Gold'
goldenMedals_filter.value_counts()

False    38607
True      2009
Name: Medal, dtype: int64

In [29]:
medals_filter = df['Medal']. value_counts()
medals_filter

Gold      2009
Bronze    1994
Silver    1914
Name: Medal, dtype: int64

In [36]:
#Ahora vamos a ver cuantos deportistas > 40 años ganaron medalla de Oro

over_40_gold_df = over_40_filter & goldenMedals_filter

over_40_gold_df.value_counts()

#66 deportistas > 40 años ganaron medalla de Oro


False    40550
True        66
dtype: int64

3. Determinen la altura promedio de:

     a. Todas las filas incluidas.

     b. Todas las filas incluidas, pero agrupadas por Event (evento).

     c. Bono: Este conjunto de datos contiene muchos atletas más de una vez (si compitieron en varios eventos y/o varios años). Pueden confirmarlo encontrarlo .value_counts en la columna ID.  Para la tarea del bono queremos que obtengan la altura promedio de todos los atletas, pero sin incluir algún atleta más de una vez.  Pista: Aprendimos cómo eliminar duplicados si CADA columna fuera idéntica, sin embargo, ahora tenemos que eliminar una fila cada vez que un ID se repite.  Para hacer eso, pueden usar un subconjunto de sus datos como un argumento en la función drop_duplicates.  También le pueden decir a Python cuales duplicados mantener usando el argumento keep.

In [42]:
# find the mean rating for Heigh across all athletes
df['Height'].mean()


175.3811165585956

In [52]:
df.groupby(['Event'])['Height'].mean()

Event
Alpine Skiing Men's Combined                      178.547170
Alpine Skiing Men's Downhill                      178.211679
Alpine Skiing Men's Giant Slalom                  176.739394
Alpine Skiing Men's Slalom                        176.482143
Alpine Skiing Men's Super G                       179.850575
                                                     ...    
Wrestling Women's Flyweight, Freestyle            158.100000
Wrestling Women's Heavyweight, Freestyle          173.444444
Wrestling Women's Light-Heavyweight, Freestyle    170.000000
Wrestling Women's Lightweight, Freestyle          162.666667
Wrestling Women's Middleweight, Freestyle         164.533333
Name: Height, Length: 726, dtype: float64

In [60]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40616 entries, 0 to 40615
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   ID      40616 non-null  int64  
 1   Age     39143 non-null  float64
 2   Height  31615 non-null  float64
 3   Team    40616 non-null  object 
 4   NOC     40616 non-null  object 
 5   Games   40616 non-null  object 
 6   Year    40616 non-null  int64  
 7   Season  40616 non-null  object 
 8   City    40616 non-null  object 
 9   Sport   40616 non-null  object 
 10  Event   40616 non-null  object 
 11  Medal   5917 non-null   object 
dtypes: float64(2), int64(2), object(8)
memory usage: 3.7+ MB
